test/Microsoft.ML.Tests/Scenarios/TensorflowTests.cs

```diff
@@ -39,7 +39,7 @@ public void TensorFlowTransforCifarEndToEndTest()
                     .Append(new ImagePixelExtractingEstimator(mlContext, "ImageCropped", "Input", interleave: true))
                     .Append(new TensorFlowEstimator(mlContext, model_location, new[] { "Input" }, new[] { "Output" }))
                     .Append(new ColumnConcatenatingEstimator(mlContext, "Features", "Output"))
-                    .Append(new ValueToKeyMappingEstimator(mlContext, "Label"))
+                    .Append(new ValueToKeyMappingEstimator(mlContext, "Label", "Label"))
```

Comment:
>, "Label" [](start = 77, length = 9)

side effect of this change is that we cannot have the output column default to null, and therefore substitute it with the input column when not provided...

Reply:
Let's reverse that. We should retain the behavior and attitude that existed before the fiasco began, of the "name" (what was misidentified as "output") always being specified, and in the case where there *is* a single "source" (what was misidentified as "input"), it is taken as being the same as the "name."

---
In reply to: [246970621](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246970621) [](ancestors = 246970621)

Reply:
This was marked as resolved, but I still see that we've retained this change.

---
In reply to: [246974415](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246974415) [](ancestors = 246974415,246970621)